docs/DeveloperGuide.adoc

```diff
@@ -267,17 +286,105 @@ For example, using a single `**List<Transaction>**` instead of needing 2 separat
+=== Setting budgets
+We allow the user to maintain a `**Budget**` for each calendar month. This section details how `**THRIFT**` handle
+requests made by the user who is trying to set a budget for their desired month. Each `**Budget**` stores a `**Calendar**`
+and a `**BudgetValue**`. The following class diagram shows how a `**Budget**` object looks like:
+
+image::add-budget/BudgetClassDiagram.png[]
+
+==== Implementation
+We store every `**Budget**` set by the user into `BudgetList#List<Budget>`, which is a list object in `**BudgetList**`.
+This list will contain only one unique `**Budget**` object for each month, with the month uniqueness indicated by `Budget#period` in the "MM/YYYY" format.
+Below is a class diagram explaining how the relations look:
+
+image::add-budget/BudgetListClassDiagram.png[]
+
+Whenever the user attempts to add a `**Budget**`, `**THRIFT**` will check if the `**Budget**` currently exists in `BudgetList#List<Budget>`.
+`**BudgetList**` contains a few methods that simplifies such management of the list:
+
+. `BudgetList#getBudgetForMonthYear(Calendar toCheck)` - Checks if `toCheck` matches any of the `Budget#period` in `List<Budget>`.
+. `BudgetList#setBudget(Budget toSet)` - Adds `toSet` into `List<Budget>` if `toSet#period` does not match any of the `Budget#period` in the list. Otherwise,
+overwrite the matching `**Budget**` in `List<Budget>` with `toSet`.
+. `BudgetList#removeBudget(Budget toRemove)` - Removes `toRemove` from `List<Budget>`.
+
+We will demonstrate what happens at the back-end whenever the user sets a budget and overwrites it afterwards. The user has not set any `**Budget**`
+before, so `BudgetList#List<Budget>` is empty:
+
+Step 1. The user wishes to set their budget for October 2019 as $1500. They execute the command: **budget v/1500 d/10/2019**.
+The user's entry is checked by `BudgetCommandParser#parse()` and an attempt to parse each parameter occurs:
+
+* `**Calendar**` is parsed by `ParseUtil#parseDate(ArgumentMultimap)`
+* `**BudgetValue**` is parsed by `ParseUtil#parseBudgetValue(ArgumentMultimap)`
+
+NOTE: `**ArgumentMultimap**` is a class that stores all the parsed parameters taken from the user input.
+
+Since the user input is valid, the `**Budget**` is successfully created and inserted into `BudgetList#List<Budget>`.
+
+Step 2. The user wishes to set a new budget for October 2019 as $2000. They execute the command: **budget v/2000 d/10/2019**.
+The user's entry is once again checked to be valid and a new `**Budget**` is created. Before it is inserted into
+`BudgetList#List<Budget>`, a check to see if a budget is already set for that month will occur. Since there is already a
+budget set for October 2019, the existing budget is replaced by this newly created `**Budget**`.
+
+The following code snippet is part of `BudgetList#setBudget(Budget toSet)` and demonstrates how this checking is done:
+
+.BudgetList.java
+[source,java]
+----
+//List<Budget> internalList = new ArrayList<>();
```

Comment:
Is this comment line added intentionally?

Reply:
Added another comment for clarity